python/lsst/pipe/base/pipeline.py

```diff
@@ -768,9 +794,15 @@ def makeDatasetTypesSet(connectionType: str, freeze: bool = True) -> NamedValueS
                     try:
                         registryDatasetType = registry.getDatasetType(c.name)
                     except KeyError:
-                        compositeName, componentName = DatasetType.splitDatasetTypeName(c.name)
-                        parentStorageClass = DatasetType.PlaceholderParentStorageClass \
-                            if componentName else None
+                        _, componentName = DatasetType.splitDatasetTypeName(c.name)
+                        if componentName:
+                            if not usePlaceHolder:
+                                raise ValueError("Component parent class cannot be determined, and "
+                                                 "placeholder was not supplied")
+                            else:
+                                parentStorageClass = PlaceHolder
```

Comment:
I'm a bit confused. I know I've been out the loop but why have we replaced an explicit placeholder definition in daf butler by a completely generic place holder that now has no special meaning at all?

Reply:
This is exactly what you asked me to do in a weekly meeting, removing this special value from daf_butler. I have no strong opinion on it, but you previously indicated you preferred it gone